server/impl/src/main/java/com/walmartlabs/concord/server/process/queue/dispatcher/Dispatcher.java

```diff
@@ -213,23 +222,26 @@ private List<Match> match(DSLContext tx, List<Request> requests) {
         return matches;
     }
 
-    static Request findRequest(ProcessQueueEntry candidate, List<Request> requests) {
+    static Request findRequest(ProcessQueueEntry candidate, List<Request> requests, RequirementsMatcherErrorHandler errHandler) {
         return requests.stream()
-                .filter(req -> isRequestMatch(candidate, req))
+                .filter(req -> isRequestMatch(candidate, req, errHandler))
                 .findFirst()
                 .orElse(null);
     }
 
-    private static boolean isRequestMatch(ProcessQueueEntry candidate, Request req) {
+    private static boolean isRequestMatch(ProcessQueueEntry candidate, Request req, RequirementsMatcherErrorHandler errHandler) {
         try {
             var capabilities = req.request.getCapabilities();
             var requirements = getAgentRequirements(candidate);
 
             return requirements.isEmpty() || Matcher.matches(capabilities, requirements);
         } catch (PatternSyntaxException pse) {
             log.error("Invalid regex in requested agent capabilities for instanceId: {}", candidate.key().getInstanceId());
+            errHandler.handle(candidate, pse);
         } catch (Exception e) {
-            log.error("Error matching process requirements: {}", e.getMessage());
+            var errMsg = String.format("Error matching process requirements for instanceId: %s", candidate.key().getInstanceId());
```

Comment:
I propose we don't mix var/no-var within one .java file.

Reply:
That's fair 👍